### PR TITLE
Avoid strack traces on OAuth /profile

### PR DESCRIPTION
**Pull request by leleuj**
Depending on the malformed JWT access token sent to the OAuth /profile endpoint, a direct stack trace can be returned.

This PR sanitizes this issue by returning a generic "invalid_request" error.

### CODE REVIEW

**Comment by leleuj** [comment]
Backported to the 6.6.x branch: https://github.com/apereo/cas/pull/5909

File: ci/tests/puppeteer/scenarios/oauth2-login-jwt-at/script.js [hunk 1]
```diff
@@ -63,6 +63,25 @@ async function executeFlow(browser, redirectUri, clientId, accessTokenSecret) {
...
 
+    const parts = accessToken.split('.');
+    let badAccessToken;
+    if (parts.length === 3) {
+        badAccessToken = parts[0] + '.Z' + parts[1] + '.' + parts[2];
+    } else {
+        badAccessToken = parts[0] + '.' + parts[1] + '.Z' + parts[2] + '.' + parts[3] + '.' + parts[4];
+    }
+    const badParams = new URLSearchParams();
+    badParams.append('access_token', badAccessToken);
```
Comment (mmoayyed) [inline ci/tests/puppeteer/scenarios/oauth2-login-jwt-at/script.js:74]: Could I ask, why are you doing this to create a bad access token? Isn't something like `kwdhfkusdyfksdbfkjdsgfhjsdgfhkjhseriuyewiurhskjfbcdkjfbdsjk` enough?

File: support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/endpoints/OAuth20UserProfileEndpointController.java [hunk 2]
```diff
@@ -67,7 +67,11 @@ public ResponseEntity<String> handlePostRequest(final HttpServletRequest request
     public ResponseEntity<String> handleGetRequest(final HttpServletRequest request,
                                                    final HttpServletResponse response) throws Exception {
         response.setContentType(MediaType.APPLICATION_JSON_VALUE);
-        val accessTokenResult = getAccessTokenFromRequest(request);
+        val accessTokenResult = FunctionUtils.doAndHandle(() -> getAccessTokenFromRequest(request));
+        if (accessTokenResult == null) {
+            LOGGER.error("Unable to get the access token from the request");
+            return buildUnauthorizedResponseEntity(OAuth20Constants.INVALID_REQUEST);
+        }
```
Comment (mmoayyed) [inline support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/endpoints/OAuth20UserProfileEndpointController.java:74]: Please make sure you have unit tests to catch these lines as well.

File: ci/tests/puppeteer/scenarios/oauth2-login-jwt-at/script.js (hunk 1 above)
Comment (mmoayyed) [reply ci/tests/puppeteer/scenarios/oauth2-login-jwt-at/script.js:74]: Or if you need it to be a jwt, why not create a proper JWT? What exists here is very black magic.

Comment (leleuj) [reply ci/tests/puppeteer/scenarios/oauth2-login-jwt-at/script.js:74]: In fact, to produce the `Unable to parse JWT` exception, I needed to change the content (not the header or signature) of the JWT.

File: support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/endpoints/OAuth20UserProfileEndpointController.java (hunk 2 above)
Comment (mmoayyed) [reply support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/endpoints/OAuth20UserProfileEndpointController.java:74]: Yes please., if you don't mind These will show as an "untested" lines without unit tests when we run coverage reports.

Comment (leleuj) [reply support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/endpoints/OAuth20UserProfileEndpointController.java:74]: OK. Will do (tomorrow).

File: ci/tests/puppeteer/scenarios/oauth2-login-jwt-at/script.js (hunk 1 above)
Comment (mmoayyed) [reply ci/tests/puppeteer/scenarios/oauth2-login-jwt-at/script.js:74]: I'd very much recommend that you add a comment here and explain what happens here then. 5 months from now, it's virtually impossible to figure out what's being tested here. Even better, you could refactor this into a "createAccessTokenThatDoesXYZ" so comments would be unnecessary.

Comment (leleuj) [reply ci/tests/puppeteer/scenarios/oauth2-login-jwt-at/script.js:74]: I don't want a proper JWT here, I want a malformed JWT. This is why I create it manually.

Comment (leleuj) [reply ci/tests/puppeteer/scenarios/oauth2-login-jwt-at/script.js:76]: I have added an explanation in the Puppeteer test.

File: support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/endpoints/OAuth20UserProfileEndpointController.java (hunk 2 above)
Comment (leleuj) [reply support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/endpoints/OAuth20UserProfileEndpointController.java:74]: I have added a unit test to confirm the behavior.

**Comment by leleuj** [comment]
@mmoayyed I have taken into account your feedbacks (both for this PR and https://github.com/apereo/cas/pull/5909).

This is ready for merge. That said, I'm not sure this still makes sense to add/modify a Puppeteer test when there is already a simpler unit test.

**Comment by leleuj** [comment]
@apereocas-bot runci

**Pull request merged**